plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,310 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t, validateNonEmpty, validateInteger } from '@superset-ui/core';
+import {
+  sharedControls,
+  ControlPanelConfig,
+  D3_FORMAT_OPTIONS,
+  ColumnOption,
+  ColumnMeta,
+} from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+
+const columnSelectControl = {
+  ...sharedControls.entity,
+  type: 'SelectControl',
+  multi: true,
+  freeForm: true,
+  default: null,
+  includeTime: false,
+  optionRenderer: (c: ColumnMeta) => <ColumnOption column={c} showType />,
+  valueRenderer: (c: ColumnMeta) => <ColumnOption column={c} />,
+  valueKey: 'column_name',
+  allowAll: true,
+  filterOption: ({ data: opt }: { data: ColumnMeta }, text = '') =>
+    opt.column_name?.toLowerCase().includes(text.toLowerCase()) ||
+    opt.verbose_name?.toLowerCase().includes(text.toLowerCase()),
+  promptTextCreator: (label: string) => label,
+  commaChoosesOption: false,
+};
+
+const config: ControlPanelConfig = {
+  controlPanelSections: [
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [
+          {
+            name: 'columns',
+            config: {
+              ...columnSelectControl,
+              clearable: false,
+              label: t('Columns'),
+              description: t('Columns to group by'),
+            },
+          },
+        ],
+        ['metric'],
+        ['adhoc_filters'],
+        ['row_limit'],
+      ],
+    },
+    {
+      label: t('Chart Options'),
+      expanded: true,
+      controlSetRows: [
+        [<h1 className="section-header">{t('General')}</h1>],
+        [
+          {
+            name: 'min_val',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              default: DEFAULT_FORM_DATA.minVal,
+              validators: [validateNonEmpty, validateInteger],
+              renderTrigger: true,
+              label: t('Min'),
+              description: t('Minimum value on the gauge axis'),
+            },
+          },
+          {
+            name: 'max_val',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              default: DEFAULT_FORM_DATA.maxVal,
+              validators: [validateNonEmpty, validateInteger],
+              renderTrigger: true,
+              label: t('Max'),
+              description: t('Maximum value on the gauge axis'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'start_angle',
+            config: {
+              type: 'TextControl',
+              label: t('Start angle'),
+              description: t('Angle at which to start progress axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.startAngle,
```

Comment:
It might be cleaner to do object destructuring and then use those variables here:

```javascript
const { startAngle, endAngle, fontSize, ... } = DEFAULT_FORM_DATA;
```

Reply:
Sorry,I asked for it because jesse suggested similar for graph pr.